Closes #51: To Do List (Journal / Daily View) - Functionality with dummy data

cng822 opened the pull request:
Closes #51 

## Proposed Changes

- Implemented functionality with dummy data (Migrate/Cancel/Delete are able to alter the To Do List)
- Added more style features

To do List view with Cancel(or Uncancel after you have previously cancelled) /Overdue/Normal/Completed examples 
![image](https://user-images.githubusercontent.com/50128761/111872917-d09d6900-89f2-11eb-9224-0c28052350f8.png)

New To-Do pops up as a modal for the user to type the description of the task and when they want the task to have a due date
![image](https://user-images.githubusercontent.com/50128761/111873114-9c767800-89f3-11eb-9331-b0d94ec2ae97.png)

The migrate modal if the user chooses to migrate the event to a different date well before the due date
![image](https://user-images.githubusercontent.com/50128761/111873131-b3b56580-89f3-11eb-80d3-3f12e4d886d5.png)

cng822 committed:
implemented design todo list

cng822 committed:
implemented design todo list

cng822 committed:
Update index.js

cng822 committed:
Update index.js

cng822 committed:
Update index.js

cng822 committed:
Update index.js

cng822 committed:
Update index.js

cng822 committed:
Update index.js

cng822 committed:
added delete/cancel features

cng822 committed:
added delete/cancel features

cng822 committed:
added dummy data

cng822 committed:
added dummy data

cng822 committed:
change syntax

cng822 committed:
change syntax

cng822 committed:
deleted function updated

cng822 committed:
deleted function updated

cng822 committed:
fixed delete

cng822 committed:
fixed delete

cng822 committed:
updated for modal

cng822 committed:
updated for modal

cng822 committed:
added modals for migrate

cng822 committed:
added modals for migrate

cng822 committed:
minor changes

cng822 committed:
minor changes

cng822 committed:
additional style changes

cng822 committed:
additional style changes

lhuijser commented on telos-frontend/src/components/todo/JournalTodo.jsx:
```diff
@@ -1,3 +1,386 @@
+          <form noValidate>
+            <TextField
+              id="date"
+              label="Due Date"
```
For the migrate, the user should be able to select the date for the task to move to, as well as the new due date, is it possible to include this? Otherwise, changing 'due date' to something like 'date to move to' would be good if that's possible, because migrating doesn't necessarily change the due date :)

cng822 replied:
Added :)
![image](https://user-images.githubusercontent.com/50128761/111888460-99a27400-8a41-11eb-8fbd-e927829ca098.png)

lhuijser reviewed (changes requested):
Looks amazing! Thank you for implementing this, you've done a great job with it :) Just one thought I've commented on before I'm happy to approve

ODE-BOY committed:
Changed the Journal Calendar colour from blue to purple (#154)

* Implement a theme and Theme Provider

Created a theme which changes the colour from blue to purple and a provider to implement the changes to on the app.

* Formatting Fix

Removed spaces before  and after the '=' in the theme provider

RawiriHohepa committed:
Fix components reaching past screen size then requiring scrolling (#164)

ysway committed:
Closes #55: Habit tracker (Journal / Daily View) - Functionality w/ dummy data (#153)

* typo fixed and hover effect

* add vs code ignore

* rm root folder package-lock.json

* Partial function

* implement functions

* Update addbutton style

* reuse sub-component and merge imports

flexzy committed:
Fixed static repeated port for express app in integration tests (#168)

cng822 committed:
added a text input for due date in migrate

cng822 committed:
added a text input for due date in migrate

RawiriHohepa reviewed (dismissed):
Can you rebase to the latest version of main and fix the merge conflicts?

cng822 committed:
resolved conflicts

cng822 committed:
changed css file to module.css

ysway commented:
Looks good for me but I still wanna know why there are so many files changed

cng822 committed:
fixed component styling issue

cng822 commented:
> Can you rebase to the latest version of main and fix the merge conflicts?

done!

cng822 commented:
> > Can you rebase to the latest version of main and fix the merge conflicts?
> 
> done!

> Looks good for me but I still wanna know why there are so many files changed

Likely due to all my files updating due to rebasing

ysway commented:
> > > Can you rebase to the latest version of main and fix the merge conflicts?
> > 
> > 
> > done!
> 
> > Looks good for me but I still wanna know why there are so many files changed
> 
> Likely due to all my files updating due to rebasing

As long as you only modified your components, it is fine for me to approve this PR

a-158w reviewed (approved):
Looks slick and the styling is working well! I'll assume most of the file changes are from the rebasing etc. LGTM!
